src/charm.py

```diff
@@ -63,10 +63,19 @@ def __init__(self, *args: typing.Any):
         self.framework.observe(
             self.on[constants.PEER].relation_joined, self._on_peer_relation_joined
         )
-        self.unit.open_port("tcp", 8080)  # ACL API
         self.unit.open_port("tcp", 53)  # Bind DNS
         self.unit.open_port("udp", 53)  # Bind DNS
 
+        # Try to check if the `charmed-bind-snap` resource is defined.
+        # Using this can be useful when debugging locally
+        # More information about resources:
+        # https://juju.is/docs/sdk/resources#heading--add-a-resource-to-a-charm
+        self.snap_path: str = ""
```

Comment:
I think the intent here is to note if the `snap_path` is not defined. An empty string is not always clear because an empty string could be a valid value. It is generally better to be explicit and make it `None` if there is no resource that is defined and only put a string if there is a resource to make the type clearly reflect whether the resource is set

Reply:
I've initialized snap_path to `None`